src/utils/createContextWrapper.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+/**
+ * Creates new trigger class that injects context into overlay.
+ */
+export default function createContextWrapper(Trigger, propName) {
+  return function (contextTypes) {
+    class ContextWrapper extends React.Component {
```

Comment:
In actual use, I would set up a module that does e.g.

```js
export default OverlayTrigger.withContext{
  router: React.PropTypes.func
});
```

and just accomplish the memoization that way.

I definitely don't want people to be calling `OverlayTrigger.withContext` every time - just do it once at module level. Not too worried about this since we're not documenting this anyway.